dbld/rules

```diff
@@ -62,8 +62,13 @@ bootstrap: bootstrap-$(DEFAULT_IMAGE)
 bootstrap-%: setup
 	$(DOCKER) run $(DOCKER_RUN_ARGS) --rm -e CONFIGURE_OPTS="$${CONFIGURE_OPTS:-$(CONFIGURE_OPTS)}" -ti  balabit/syslog-ng-$* /source/dbld/bootstrap
 
+make: setup-$(DEFAULT_IMAGE)
 make-%: setup
-	$(DOCKER) run $(DOCKER_RUN_ARGS) --rm -ti  balabit/syslog-ng-$(DEFAULT_IMAGE) /source/dbld/make $(MAKE_ARGS) $*
+	$(DOCKER) run $(DOCKER_RUN_ARGS) --rm -ti  balabit/syslog-ng-$* /source/dbld/make $(MAKE_ARGS)
+
```

Comment:
hm. with this change I would be unable to run make-check and make-install. I am not against changing it, but I don't see how this functionality would be available otherwise?

Previously make only supported specifying the image by supplying DEFAULT_IMAGE on the make command line, e.g.

./dbld/rules make-check DEFAULT_IMAGE=xenial

The goals behind the targets were:
* image related targets expected the image name as a part of the target (e.g. image-xenial or image-centos6)
* development related targets (bootstrap, release, etc) used the DEFAULT_IMAGE variable, as it is expected that an interactive build would always happen in the same environment, that can even be specified in ./dbld/rules.conf
* to still override the image, one could use the DEFAULT_IMAGE override, e.g.

```
$ ./dbld/rules bootstrap DEFAULT_IMAGE=xenial
$ ./dbld/rules make-check DEFAULT_IMAGE=xenial
```

In hindsight, I am not entirely sure the Makefile format was such a good idea, especially as it was using shell scripts for more complicated problems anyway, so that it could run them easily under docker.

Reply:
Thanks for the notes about the image related targets and developement related targets.
I am answering in a last comment.